### ISSUE TITLE
APIM named value creation fails for Key Vault value on the first run

### ISSUE THREAD

**Issue by SergiyDivnych** [hashicorp/terraform-provider-azurerm#19255]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.4

### AzureRM Provider Version

3.30.0

### Affected Resource(s)/Data Source(s)

azurerm_api_management_named_value

### Terraform Configuration Files

```hcl
resource "azurerm_api_management_named_value" "key_vault_named_value" {
  for_each            = var.key_vault_named_values
  name                = each.key
  resource_group_name = var.resource_group_name
  api_management_name = azurerm_api_management.api_management.name
  display_name        = each.key
  secret              = true
  value_from_key_vault {
    secret_id = each.value
  }
}
```

### Debug Output/Panic Output

```shell
│ Error: waiting on creating/updating Named Value: (Name "Clientid" / Service Name "apimxxxxxx" / Resource Group "rg-xxx-xxx-westus3"): Future#WaitForCompletion: the number of retries has been exceeded: StatusCode=400 -- Original Error: Code="ValidationError" Message="The remote name could not be resolved: 'kv-xxx-xxx-xxx.vault.azure.net'"
│ 
│   with module.api_management.azurerm_api_management_named_value.key_vault_named_value["Clientid"],
│   on baseline_modules/api_management/main.tf line 75, in resource "azurerm_api_management_named_value" "key_vault_named_value":
│   75: resource "azurerm_api_management_named_value" "key_vault_named_value" {
```

### Expected Behaviour

Named value with Key Vault reference should be added in API management

### Actual Behaviour

It's failing for the first run creation, but after second run it will be created. I'm running full infrastructure deployment, and Key Vault with all secrets created beforehand, with assigned access policies for APIM in Key Vault to Get and List secrets, so it should be able to create named values with Key Vault reference. I didn't have such problem in September with 3.20.0 provider with the same configuration.

**Comment by ziyeqf** [comment]
Hi @SergiyDivnych, thanks for opening the issue.

May I ask if your `azurerm_api_management` resource and `azurerm_key_vault` are in different locations? this information may help us investigate.

Thank you!

**Comment by SergiyDivnych** [comment]
Hi @ziyeqf. Both in the same region, and resource group.

**Comment by ziyeqf** [comment]
@SergiyDivnych Thanks for your information. 

Assuming it's related to RP of the reousrce, raised an issue on azure rest api issue https://github.com/Azure/azure-rest-api-specs/issues/21655 .

**Comment by katbyte** [comment]
Thank you for taking the time to report this issue, and apologies that it has sat without a resolution for so long.

This bug was reported against an early major version of the AzureRM provider. The provider has since moved on to v5.x, and the affected functionality has often been significantly reworked across those major releases, so bug reports from legacy major versions prior to v4.0 are generally no longer actionable as written.

As part of a cleanup of historical issues we are closing bug reports that target legacy provider versions. If you are still running into this behaviour on the current major version, please open a new issue with up-to-date reproduction details (feel free to reference this one) so it can be triaged with current context.

Thank you again for the report, and for your patience.
